File: max-subarray.py
# https://leetcode.com/problems/maximum-subarray/submissions/

class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxSum = nums[0]
        possSum = 0
        if len(nums)==1:
            return nums[0]
        for i in range(len(nums)):
            if possSum < 0:
                possSum = 0  
            possSum += nums[i]
            if maxSum < possSum:
                maxSum = possSum
        return maxSum

File: max-subarray.js
/* https://leetcode.com/problems/maximum-subarray/submissions/ */

var maxSubArray = function (nums) {
	let maxSum = nums[0];
	let possSum = 0;
	count = 0;
	if (nums.length === 1) {
		return maxSum;
	}
	for (let i = 0; i < nums.length; i++) {
		count += 1;
		if (possSum < 0) {
			possSum = 0;
		}
		possSum += nums[i];
		if (possSum > maxSum) {
			maxSum = possSum;
		}
	}
	return maxSum;
};
